scripts/shared/cloud-prepare.sh

```diff
@@ -28,8 +28,30 @@ function check_gateway_exists() {
 }
 
 function prepare_kind() {
-    read -r -a nodes <<< "${cluster_nodes[$cluster]}"
-    kubectl label node "${cluster}-${nodes[-1]}" "${GATEWAY_LABEL}" --overwrite
+    read -r -a nodes <<< "${label_gw_nodes[$cluster]}"
+
+    worker_count=$(grep -o worker <<< "${nodes[*]}" | wc -l || true)
+    master_count=$(grep -o control-plane <<< "${nodes[*]}" | wc -l || true)
+
```

Comment:
I think we should default to `label-gateway: worker`, ie if `worker_count==0 && master_count==0 ` then `worker_count=1` to match the previous behavior and to avoid changing YAML in every project.

Reply:
In the Slack discussion @skitt mentioned that he prefers to be able to define default parameter.
https://coreos.slack.com/archives/C0134E73VH6/p1663674120818639

So, according to your comment, I can suggest that we would leave the `label-gateway` parameter for the `.shipyard.e2e*` global definition, but it will not be mandatory:
```
---
submariner: true
nodes: control-plane worker
label-gateway: worker
clusters:
  cluster1:
  cluster2:
```

So if that parameter is not defined, ex.:
1)
```
---
submariner: true
nodes: control-plane worker
clusters:
  cluster1:
  cluster2:
```
or
2)
```
---
submariner: true
nodes: control-plane worker
clusters:
  cluster1:
  cluster2:
    nodes: control-plane worker worker
    label-gateway: worker worker
```

then, a single worker node as a gateway will be applied on all clusters in first example and on first cluster in second example.

@tpantelis 
What do you think?

Reply:
Ok, let me clarigy a bit about my above comment.

If the `'label-gateway: worker` parameter is not defined on global or per cluster level, a single worker gateway node will be applied.
But if label parameter applied on global level, it will be used by all the clusters.
And if label parameter applied on specific cluster, this cluster will use the defined config and rest of the clusters will apply single worker node.

Is it acceptable?

Reply:
That sounds reasonable.

Reply:
Exactly.
Will work on it.

Reply:
Done

Reply:
Just to be clear, current default is labeling the last node (doesnt matter if its control-plane or worker).